baseplate/file_watcher.py

```diff
@@ -32,49 +32,69 @@
 
 import logging
 import os
+import typing
+
+from typing import TypeVar, Generic, Callable, Optional, IO, Union, Type, NamedTuple
 
 from baseplate.retry import RetryPolicy
 
 
 logger = logging.getLogger(__name__)
 
 
-_NOT_LOADED = object()
+class _NOT_LOADED:
+    pass
```

Comment:
I went down a rabbit hole looking at this change. It looks like they recommend using an enum for these singleton instances since they can't be subclassed:
https://github.com/python/typing/pull/240/files

I like the way you've done it more tbh, but wanted to point out that the pep recommends something else. I'm totally okay with what you've got, but wanted to share

Reply:
Ah yeah, I saw the enum thing. But [this comment](https://github.com/python/typing/issues/236#issuecomment-232554039) linked to an implementation ticket in mypy that is still open: https://github.com/python/mypy/issues/1803

so i think it doesn't work yet? didn't actually try because I'm a jerk. Would you prefer I give the proper way a shot?

Reply:
Yours feels cleaner and I prefer it. I don't think anyone is going to subclass this `_NOT_LOADED` class and get confused.